lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -723,14 +724,18 @@ def validate_ip(module, cidr_ip):
     split_addr = cidr_ip.split('/')
     if len(split_addr) == 2:
         # this_ip is a IPv4 or IPv6 CIDR that may or may not have host bits set
-        # Get the network bits.
+        # Get the network bits if IPv4, and validate if IPv6.
         try:
             ip = to_subnet(split_addr[0], split_addr[1])
+            if ip != cidr_ip:
+                module.warn("One of your CIDR addresses ({0}) has host bits set. To get rid of this warning, "
+                            "check the network mask and make sure that only network bits are set: {1}.".format(cidr_ip, ip))
         except ValueError:
-            ip = to_ipv6_subnet(split_addr[0]) + "/" + split_addr[1]
-        if ip != cidr_ip:
-            module.warn("One of your CIDR addresses ({0}) has host bits set. To get rid of this warning, "
-                        "check the network mask and make sure that only network bits are set: {1}.".format(cidr_ip, ip))
+            # IPv6 works a little differently. We don't want to modify the cidr_ip, just make sure it's valid.
+            if not isinstance(ip_network(cidr_ip), IPv6Network):
+                # TODO: warn or fail?  probably better to let aws decide what it'll accept?
```

Comment:
I'd prefer to warn and let AWS decide what it will accept. If AWS also modifies some IPv6 addresses I don't think we can fail without adding a regression. Ideally we could intercept those changes as we do for IPv4 to accurately reflect what will happen, but if not, I think warning about the lack of idempotent behavior would be helpful.